File: src/main/java/com/gitlab/lae/intellij/actions/TextAction.java
package com.gitlab.lae.intellij.actions;

import com.intellij.openapi.editor.actions.TextComponentEditorAction;

class TextAction extends TextComponentEditorAction {

    TextAction(boolean write, Edit edit) {
        super(write ? edit.toWriteHandler() : edit.toHandler());
    }

}

File: src/main/kotlin/com/gitlab/lae/intellij/actions/TextAction.kt
package com.gitlab.lae.intellij.actions

import com.intellij.openapi.editor.actionSystem.EditorActionHandler
import com.intellij.openapi.editor.actions.TextComponentEditorAction

internal abstract class TextAction(
        text: String,
        handler: EditorActionHandler
) : TextComponentEditorAction(handler) {

    init {
        templatePresentation.text = text
    }

    val id: String get() = javaClass.name

}
